main.py

```diff
@@ -2217,55 +2250,143 @@ def initialize_triggers(self):
+        # FIXME: This is somewhat language-specific to look for an opening parenthesis. Most languages use parentheses
+        # for their parameter lists though. Do we make a bunch of if-elif cases for the exceptional languages that don't
+        # use parentheses?
+        index = signature.find('(')
+        string = signature[index + 1:]
+        return signature[:index + 1] + re.sub(pattern, replacement, string, 1)
 
     def handle_response(self, response, point):
-        if response is not None:
-            config = config_for_scope(self.view)
-            signatures = response.get("signatures")
-            activeSignature = response.get("activeSignature")
-            debug("got signatures, active is", len(signatures), activeSignature)
-            if len(signatures) > 0 and config:
-                signature = signatures[activeSignature]
-                debug("active signature", signature)
-                formatted = []
-                formatted.append(
-                    "```{}\n{}\n```".format(config.languageId, signature.get('label')))
-                params = signature.get('parameters')
-                if params:
-                    for parameter in params:
-                        paramDocs = parameter.get('documentation')
-                        if paramDocs:
-                            formatted.append("**{}**\n".format(parameter.get('label')))
-                            formatted.append("* *{}*\n".format(paramDocs))
-
-                formatted.append(signature.get('documentation'))
-
-                mdpopups.show_popup(
-                    self.view,
-                    preserve_whitespace("\n".join(formatted)),
-                    css=".mdpopups .lsp_signature { margin: 4px; } .mdpopups p { margin: 0.1rem; }",
-                    md=True,
-                    flags=sublime.HIDE_ON_MOUSE_MOVE_AWAY,
-                    location=point,
-                    wrapper_class="lsp_signature",
-                    max_width=800)
+        if not response:
+            return
+        config = config_for_scope(self.view)
+        if not config:
+            debug("no config found for view", self.view.id())
+            return
+
+        # Fetch all the relevant data.
+        self.signatures = response.get("signatures", None)
+        x, y = self.view.viewport_extent()
+        if self.signatures:
+            self.active_signature = response.get("activeSignature", -1)  # Optional.
+            self.active_parameter = response.get("activeParameter", -1)  # Optional.
+            self.view.show_popup(self._build_popup_content(), 0, point, int(x), int(y),
+                                 self._on_navigate, self._on_hide)
+            self._visible = True
+
+    def _on_navigate(self, url):
+        sublime.run_command("open_url", {"url": url})
+
+    def _on_hide(self):
+        self._visible = False
+
+    def on_query_context(self, key, _, operand, __):
```

Comment:
you really like the use of `on_query_context` over the use of `ViewCommand` or `WindowCommand`.

Reply:
The EventListener is there anyway, might as well override that method ;)
Otherwise, I'd have to look up the class.